src/app/debug/display/display.component.ts

```diff
@@ -13,14 +15,21 @@ import { CustomEditorComponent } from '../../custom-editor/custom-editor.compone
 export class DisplayComponent {
   displayReport: boolean = false;
   report: any = {};
-  @Input() currentView: any = {};
   @Output() closeReportEvent = new EventEmitter<any>();
   @ViewChild('editorComponent') editorComponent!: CustomEditorComponent;
   @ViewChild(DisplayTableComponent)
   displayTableComponent!: DisplayTableComponent;
   metadataTableVisible: boolean = false;
   @ViewChild('container') container!: ElementRef;
   @Input() containerHeight!: number;
+  @Input() currentView: View = {
+    metadataLabels: [],
+    defaultView: false,
+    crudStorage: false,
+    metadataNames: [],
+    nodeLinkStrategy: '',
+    storageName: '',
```

Comment:
You're setting these attributes to an empty array or string. Is this necessary or should the model maybe changed so that the attributes can be undefined? I've seen you do this in other places as well.

Reply:
The reason i did it this way is because if i make them able to be undefined in the interface, there'll be many errors complaining that the value could be undefined. Should i fix this by having those places first check if they have a value or is there a better way of doing this?

Reply:
Yes it's better to check whether the value is (not) undefined than to set the value to an empty string or array.

Reply:
This has been fixed, instead of setting the properties to empty arrays/strings it now checks whether they have a value or not.